### PR TITLE
chore: dependency updates

### PR DESCRIPTION
**Pull request by gsingh1**
* Updated all backend-related dependencies
* Upgraded Serverless framework to v3 and all lambda runtimes to `nodejs16.x`
* Removal of `use strict`
* Refactored dependency usages to be compatible major semver upgrade
* Set CodeBuild environment for api and e2e testing jobs

### CODE REVIEW

File: cicd/stacks/pipeline-stack.ts [hunk 1]
```diff
@@ -105,7 +102,7 @@ export class PipelineStack extends Stack {
         version: '0.2',
         phases: {
           install: {
-            commands: ['cd cicd', 'npm ci']
+            commands: [`n ${NODE_VERSION}`, 'cd cicd', 'npm ci']
```
Comment (gsingh1) [inline cicd/stacks/pipeline-stack.ts:105]: Ensure node 16 is installed before running the jobs

File: slic-tools/middy-util.js [hunk 2]
```diff
@@ -19,10 +17,15 @@ function middify (exports, options = {}) {
           log.debug({ request }, 'request')
         }
       }))
-      .use(httpEventNormalizer())
-      .use(httpJsonBodyParser())
-      .use(httpCors())
-      .use(httpErrorHandler())
+
+    if (options.isHttpHandler) {
```
Comment (gsingh1) [inline slic-tools/middy-util.js:21]: Latest version of the middy middlewares will throw if the event payload is not recognised as a http event so a new option `isHttpHandler` is introduced to conditionally add the related middlewares

**Review by direnakkocdemir — COMMENTED** [review]
Why do we remove usage of **_'use strict'_** ?

**Review by direnakkocdemir — APPROVED** [review]

**Comment by gsingh1** [comment]
> Why do we remove usage of **_'use strict'_** ?

This was removed since we are using webpack and babel to transpile and minify the code down in the lambda artefacts so that will already ensure that the strict mode is used.

**Review by eoinsha — APPROVED** [review]
LGTM 🏅

**Review by direnakkocdemir — APPROVED** [review]
Looks great!

**Pull request merged**